### ISSUE TITLE
C Assertion of code builder

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#247]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/242) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Getting this when I use `(text json schema)`:
```
ASSERT failure /home/takashi/projects/sagittarius/src/closure.c:52: SG_CODE_BUILDERP(code)
```
It seems cache of procedure contained list (but not sure)

NOTE: The actual type of `code` is SgSharedRef object.

Reproducing code:
```scheme
(library (lib1)
  (export +closures+)
  (import (rnrs))

(define (foo e)
  (unless (string? e) (assert-violation 'foo "string" e))
  (lambda (v) (string=? v e)))

(define +closures+ `(,foo))
)

;; lib2.scm
(library (lib2)
  (export bar buz)
  (import (rnrs)
          (lib1))

(define (bar s) ((buz) s))
(define (buz) (car +closures+))
)

;; test.scm
(import (rnrs) (lib2))

((bar "s") "s")
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

It seems caused by over optimisation. (The procedure contains lifted lambda and that's in the other cache file only).

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Handling constant list containing closure (Fixes #242)

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

It still happens...

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Checking closure on bytecode as well during cache reading (Fixes #242)
